extensions/amp-form/0.1/amp-form.js

```diff
@@ -626,7 +626,9 @@ export class AmpForm {
    * @private
    */
   doActionXhr_() {
-    return this.doXhr_(dev().assertString(this.xhrAction_), this.method_);
+    // Prefer decorated linker url if it exists, otherwise use action-xhr.
+    const url = this.getXhrUrl_('linker-xhr') || this.xhrAction_;
```

Comment:
can we avoid this reverse dependency of linker?

Reply:
Refactored form submit service to also pass the setter. This avoids the reverse dependency concern.